src/module/canvas/perception/modes.ts

```diff
@@ -152,12 +152,57 @@ class DetectionModeTremorPF2e extends DetectionModeTremor {
     }
 }
 
+class ThoughtsDetectionMode extends DetectionMode {
+    constructor() {
+        super({
+            id: "senseThoughts",
+            label: "PF2E.Actor.Creature.Sense.Type.Thoughtsense",
+            type: DetectionMode.DETECTION_TYPES.OTHER,
+            walls: false,
+            angle: false,
+        });
+    }
+
+    static override getDetectionFilter(): OutlineOverlayFilter {
+        const filter = (this._detectionFilter ??= OutlineOverlayFilter.create({
+            wave: true,
+            knockout: false,
+            outlineColor: [0, 1, 0, 1],
+        }));
+        filter.thickness = 1;
+        return filter;
+    }
```

Comment:
This needs to depend on acuity, showing the token in full if the sense is precise.

Reply:
getDetectionFilter() is a static function, so it isn't provided with the specific token providing the vision.